### DISCUSSION TITLE
Where can the grpc dependent files be found or is there any other way to implement them

### DISCUSSION THREAD

**Discussion by CloudWise-Lukemiao** [prometheus/prometheus#9548] [Q&A]
We are implementing a new database. We want to access the Prometheus data into this database. I checked the data and need to use remote.proto and types.proto to generate grpc files. But I found that these two files rely on gogoproto / gogo.proto, and I can't find this file. Where is this file or is there any other jar directly used?
![image](https://user-images.githubusercontent.com/76942485/137087465-bf633e87-8b67-4b07-aed1-143589bb4362.png)

**Comment by cstyan** [comment]
Gogo proto is an alternate protobuf implementation we use: https://github.com/gogo/protobuf

**Comment by bboreham** [comment]
More recently there are suggestions to use the Buf registry to generate; see https://github.com/prometheus/prometheus/issues/11908
